signer/contentsignature/contentsignature.go

```diff
@@ -108,7 +107,11 @@ func (s *ContentSigner) SignData(input []byte, options interface{}) (signer.Sign
 	}
 	alg, hash := makeTemplatedHash(input, s.Mode)
 	sig, err := s.SignHash(hash, options)
-	sig.(*verifier.ContentSignature).HashName = alg
+
+	if sig != nil {
+		// sig could be nil when there is misconfigured permission or the kms is not available
+		sig.(*verifier.ContentSignature).HashName = alg
+	}
```

Comment:
So, I have a few different bits of feedback for this.

The first: Only use objects after you're sure the error returned with them is nil. So, instead of doing this sig check, return early if `err != nil`. That err check and early return is the change that needs to happen for this patch to be shippable.

Second, doing nil checks against interfaces generally don't go well because people misinterpret their meaning and assume things are safe about the interface after that check that are not true. See https://trstringer.com/go-nil-interface-and-interface-with-nil-concrete-value/ and https://go.dev/tour/methods/13. In case be fine in cases, but it's a pattern we don't want to support and I don't think this change is shippable with it. 

I had some other points about what you could do if backed into a corner with nil interface checks, but if you were, other things would be different and my advice would be, too.

Reply:
Here's a related problem we had to fix up this week: https://github.com/mozilla-services/autograph/pull/1034

Reply:
Thanks for this info @jmhodges, I was unfamiliar with interface and happy to learn more about it. Resolved [7b19fee](https://github.com/mozilla-services/autograph/pull/1033/commits/7b19fee2a2e3daac50d64af473f12aa4ea1f72c8)